lambdas/data-migration2/tests/test-pdrs.js

```diff
@@ -0,0 +1,344 @@
+
+const testDbName = `data_migration_2_${cryptoRandomString({ length: 10 })}`;
+
+test.before(async (t) => {
+  process.env.stackName = cryptoRandomString({ length: 10 });
+  process.env.system_bucket = cryptoRandomString({ length: 10 });
+
+  process.env.CollectionsTable = cryptoRandomString({ length: 10 });
+  process.env.ProvidersTable = cryptoRandomString({ length: 10 });
+  process.env.PdrsTable = cryptoRandomString({ length: 10 });
+
+  await createBucket(process.env.system_bucket);
+
+  collectionsModel = new Collection();
+  await collectionsModel.createTable();
+
+  providersModel = new Provider();
+  await providersModel.createTable();
+
+  pdrsModel = new Pdr();
+  await pdrsModel.createTable();
+
+  t.context.pdrPgModel = new PdrPgModel();
+
+  const { knexAdmin, knex } = await generateLocalTestDb(
+    testDbName,
+    migrationDir
+  );
+  t.context.knexAdmin = knexAdmin;
+  t.context.knex = knex;
+});
+
+test.beforeEach(async (t) => {
+  const collectionPgModel = new CollectionPgModel();
+  t.context.testCollection = fakeCollectionRecordFactory();
+
+  const collectionResponse = await collectionPgModel.create(
+    t.context.knex,
+    t.context.testCollection
+  );
+  t.context.collectionCumulusId = collectionResponse[0];
+  t.context.collectionPgModel = collectionPgModel;
+
+  const providerPgModel = new ProviderPgModel();
+  t.context.testProvider = fakeProviderRecordFactory();
+
+  const providerResponse = await providerPgModel.create(
+    t.context.knex,
+    t.context.testProvider
+  );
+  t.context.providerCumulusId = providerResponse[0];
+  t.context.providerPgModel = providerPgModel;
+});
+
+test.afterEach.always(async (t) => {
+  await t.context.knex(tableNames.pdrs).del();
+  await t.context.knex(tableNames.providers).del();
+  await t.context.knex(tableNames.collections).del();
+});
+
+test.after.always(async (t) => {
+  await recursivelyDeleteS3Bucket(process.env.system_bucket);
+
+  await providersModel.deleteTable();
+  await collectionsModel.deleteTable();
+  await pdrsModel.deleteTable();
+
+  await destroyLocalTestDb({
+    ...t.context,
+    testDbName,
+  });
+});
+
+test.serial('migratePdrRecord correctly migrates PDR record', async (t) => {
+  const {
+    collectionCumulusId,
+    knex,
+    pdrPgModel,
+    providerCumulusId,
+    testCollection,
+    testProvider,
+  } = t.context;
+  const executionPgModel = new ExecutionPgModel();
+  const execution = fakeExecutionRecordFactory();
+
+  const executionResponse = await executionPgModel.create(
+    knex,
+    execution
+  );
+  const executionCumulusId = executionResponse[0];
+
+  const testPdr = generateTestPdr(testCollection.name, testProvider.name, execution.arn);
+  await migratePdrRecord(testPdr, knex);
+
+  const record = await pdrPgModel.get(knex, { name: testPdr.pdrName });
+
+  t.like(
+    record,
+    omit({
+      name: testPdr.pdrName,
+      provider_cumulus_id: providerCumulusId,
+      collection_cumulus_id: collectionCumulusId,
```

Comment:
I'm curious/surprised this is working, if it is, since the logic for looking up the PDR collection in the actual migration code seems a bit off

Reply:
I think it is working because you're passing in a collection name to set the `collectionId` value for the Dynamo record. So then that the value is used to look up the collection by `name` in PG, which works. But in reality, `collectionId` is a concatenation of name/version, like `name___version`, so the tests need to be updated to reflect that

Reply:
Yeah, I was curious about that too after seeing your other comment and I think your conclusion is correct. I've since updated the tests